File: backend/routes/order_routes.py
from flask import Blueprint, request, jsonify
from services.order_service import create_order, confirm_order, get_customer_orders

order_bp = Blueprint("orders", __name__)

@order_bp.post("/create")
def create():
    data = request.get_json()
    customer_id = data.get("customer_id")
    items = data.get("items")

    if not customer_id or not items:
        return jsonify({"error": "customer_id and items are required"}), 400

    response, status = create_order(customer_id, items)
    return jsonify(response), status


@order_bp.post("/confirm/<order_id>")
def confirm(order_id):
    response, status = confirm_order(order_id)
    return jsonify(response), status


@order_bp.get("/history/<customer_id>")
def history(customer_id):
    orders = get_customer_orders(customer_id)
    return jsonify(orders), 200

File: backend/routes/order_routes.js
const express = require('express');
const router = express.Router();
const { createOrder, confirmOrder, getCustomerOrders } = require('../services/order_service');

router.post("/create", async (req, res) => {
  const { customer_id, items } = req.body;
  
  if (!customer_id || !items) {
    return res.status(400).json({ error: "customer_id and items are required" });
  }
  
  const [response, status] = await createOrder(customer_id, items);
  return res.status(status).json(response);
});

router.post("/confirm/:order_id", async (req, res) => {
  const [response, status] = await confirmOrder(req.params.order_id);
  return res.status(status).json(response);
});

router.get("/history/:customer_id", async (req, res) => {
  const orders = await getCustomerOrders(req.params.customer_id);
  return res.status(200).json(orders);
});

module.exports = router;
